handsontable/src/dataMap/dataMap.js

```diff
@@ -596,14 +596,21 @@ class DataMap {
    * Add/remove row(s) to/from the data source.
    *
    * @param {number} index Physical index of the element to add/remove.
-   * @param {number} amount Number of rows to add/remove.
+   * @param {number} deleteCount Number of rows to remove.
    * @param {...object} elements Row elements to be added.
    */
-  spliceData(index, amount, ...elements) {
-    const continueSplicing = this.instance.runHooks('beforeDataSplice', index, amount, elements);
+  spliceData(index, deleteCount, ...elements) {
+    const continueSplicing = this.instance.runHooks('beforeDataSplice', index, deleteCount, elements);
 
     if (continueSplicing !== false) {
-      this.dataSource.splice(index, amount, ...elements);
+      const newData = [...this.dataSource.slice(0, index), ...elements, ...this.dataSource.slice(index)];
+
+      // We try not to change the reference.
+      this.dataSource.length = 0;
+
+      // Pushing to array instead of using `splice`, because Babel changes the code to one that uses the `apply` method.
+      // The used method was cause of the problem described within #7840.
+      newData.forEach(row => this.dataSource.push(row));
```

Comment:
@wszymanski Have you checked if it does affect the performance in any way?

Reply:
As I wrote in PR description:

> I measured the performance for past and present solution and it seems that there is no significant difference in performance (the measured deviation was equal to 0,7%). It's worth to mention also that there is no such problem with analogous insert_col option - it doesn't use spliceData method internally.

🙂

Reply:
Oh, my bad, I missed that part 🙇